solidity/src/Base7683.sol

```diff
@@ -0,0 +1,389 @@
+    /// @dev Intended to improve standardized integration of various order types and settlement contracts
+    /// @param order The GaslessCrossChainOrder definition
+    /// NOT USED originFillerData Any filler-defined data required by the settler
+    /// @return resolvedOrder ResolvedCrossChainOrder hydrated order data including the inputs and outputs of the order
+    function resolveFor(
+        GaslessCrossChainOrder calldata order,
+        bytes calldata
+    )
+        public
+        view
+        returns (ResolvedCrossChainOrder memory resolvedOrder)
+    {
+        (resolvedOrder,) =
+            _resolvedOrder(order.orderDataType, order.user, order.orderData, order.openDeadline, order.fillDeadline);
+    }
+
+    /// @notice Resolves a specific OnchainCrossChainOrder into a generic ResolvedCrossChainOrder
+    /// @dev Intended to improve standardized integration of various order types and settlement contracts
+    /// @param order The OnchainCrossChainOrder definition
+    /// @return resolvedOrder ResolvedCrossChainOrder hydrated order data including the inputs and outputs of the order
+    function resolve(OnchainCrossChainOrder calldata order)
+        public
+        view
+        returns (ResolvedCrossChainOrder memory resolvedOrder)
+    {
+        (resolvedOrder,) = _resolvedOrder(
+            order.orderDataType,
+            msg.sender,
+            order.orderData,
+            type(uint32).max, // there is no open deadline for onchain orders since the user is opening it the order
+            order.fillDeadline
+        );
+    }
+
+    /// @notice Fills a single leg of a particular order on the destination chain
+    /// @param _orderId Unique order identifier for this order
+    /// @param _originData Data emitted on the origin to parameterize the fill
+    /// NOT USED fillerData Data provided by the filler to inform the fill or express their preferences
+    function fill(bytes32 _orderId, bytes calldata _originData, bytes calldata) external virtual {
+        OrderData memory orderData = OrderEncoder.decode(_originData);
+
+        if (_orderId != _getOrderId(orderData)) revert InvalidOrderId();
+        if (block.timestamp > orderData.fillDeadline) revert OrderFillExpired();
+        if (orderData.destinationDomain != _localDomain()) revert InvalidOrderDomain();
+        _mustHaveRemoteCounterpart(orderData.originDomain);
+        if (orderStatus[_orderId] != OrderStatus.UNFILLED) revert InvalidOrderStatus();
+
+        orders[_orderId] = orderData;
+        orderStatus[_orderId] = OrderStatus.FILLED;
+        orderFiller[_orderId] = msg.sender;
+
+        emit Filled(_orderId, _originData, new bytes(0));
+
+        IERC20(TypeCasts.bytes32ToAddress(orderData.outputToken)).safeTransferFrom(
+            msg.sender, TypeCasts.bytes32ToAddress(orderData.recipient), orderData.amountOut
+        );
+    }
+
+    function settle(bytes32[] calldata _orderIds, address[] calldata receivers) external payable {
```

Comment:
Nice to have: It might be nice to have some comments that show

settle => _handleSettlement => _settleOrder
refund => _handleRefund => _refundOrder

and then some indicate what what a subclass should do (i.e. implement _handleSettlement and call `settleOrder` on origin)